.github/workflows/test.yml

```diff
@@ -1,31 +1,148 @@
-# This workflow will install Python dependencies, run tests and lint with a single version of Python
-# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
-
 name: Run Tests
 
+# Explicitly set permissions to allow Dependabot workflow runs to write in the PR
+# for coverage's reporting.
+# By default, these are read-only when the actions are ran by Dependabot
+# https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#changing-github_token-permissions
+permissions: write-all
+
+env:
+  ALL_PYTHON_VERSIONS: "['3.10', '3.11']"
+
 on:
   push:
     branches: [master]
   pull_request:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
 jobs:
-  build:
+  info:
+    runs-on: ubuntu-latest
+    name: Collect information & changes data
+    outputs:
+      run_pipeline: ${{ steps.changes.outputs.core }}
+      python_versions: ${{ steps.info.outputs.python_versions }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3.5.3
+
+      - name: Filter for core changes
+        uses: dorny/paths-filter@v2.11.1
+        id: changes
+        with:
+          filters: |
```

Comment:
We don't seem to need the filter as we always run either all jobs or no jobs, besides the info job. Also, this project isn't large enough to justify a filter I think.

Caching the environment setup is good as installing the requirements without cache takes a minute.

Reply:
A valid point, I'll remove the filter.